### ISSUE TITLE
Create a new intrinsic to let the user unroll a specific loop

### ISSUE THREAD

**Issue by rafaelsamenezes** [esbmc/esbmc#526]
Right now, the only way for a user to define the bound of a specific loop is by manually finding and setting it through our CLI. I suggest adding the intrinsic `__ESBMC_goto_unroll(const size_t N)` which will unroll the loop N times.

This function will affect the GOTO construction, so it cannot be replaced with assume statements.

```c
void foo() {
  while(non_det()) {
   __ESBMC_goto_unroll(3);
    bar();
  }
}

void foo_converted()
{
  bar();
  bar();
  bar();
}
```

**Comment by feliperodri** [comment]
Wouldn't be better to specify a bound without adding extra annotations to the program? You could have different verification runs with different bounds, without modifying annotations. CBMC uses mangled names for each loop, which takes into a account source location (e.g., function). The user specifies which loops would be unwind using the `--unwindset` option. Take a look at this class https://github.com/diffblue/cbmc/blob/e7eff1ea6ea08d4e5e1bf60850ca029bb9b221a7/src/goto-instrument/unwindset.h#L22.

**Comment by mikhailramalho** [comment]
> Wouldn't be better to specify a bound without adding extra annotations to the program? You could have different verification runs with different bounds, without modifying annotations. CBMC uses mangled names for each loop, which takes into a account source location (e.g., function). The user specifies which loops would be unwind using the `--unwindset` option. Take a look at this class https://github.com/diffblue/cbmc/blob/e7eff1ea6ea08d4e5e1bf60850ca029bb9b221a7/src/goto-instrument/unwindset.h#L22.

I agree, and we used to have a `--unwindset` option, however, it uses the numbering generated during goto conversion to indicate the loop, which is usually not that straightforward.

**Comment by rafaelsamenezes** [comment]
No longer valid.
